File: src/features/typing/data/go/longest_consecutive_sequence.go
package main

func longestConsecutive(nums []int) int {
	set := make(map[int]bool, len(nums))
	for _, num := range nums {
		set[num] = true
	}
	best := 0
	for num := range set {
		if set[num-1] {
			continue
		}
		length := 1
		for set[num+length] {
			length++
		}
		if length > best {
			best = length
		}
	}
	return best
}

File: src/features/typing/data/ts/longest_consecutive_sequence.ts
function longestConsecutive(nums: number[]): number {
  const set = new Set(nums);
  let best = 0;
  for (const num of set) {
    if (set.has(num - 1)) {
      continue;
    }
    let length = 1;
    while (set.has(num + length)) {
      length++;
    }
    best = Math.max(best, length);
  }
  return best;
}
